frame/support/src/traits/tokens/misc.rs

```diff
@@ -189,10 +189,11 @@ impl<
 {
 }
 
-/// Converts a balance value into an asset balance.
+/// Provides a conversion mechanism between two balances.
+/// The most natural conversion would be native to asset balance.
 pub trait BalanceConversion<InBalance, AssetId, OutBalance> {
 	type Error;
-	fn to_asset_balance(balance: InBalance, asset_id: AssetId) -> Result<OutBalance, Self::Error>;
+	fn convert(balance: InBalance, asset_id: AssetId) -> Result<OutBalance, Self::Error>;
```

Comment:
I think what you are doing in https://github.com/paritytech/substrate/pull/13608 PR, is different from what this meant for.
Here, the assets id, is id to which the in_balance should be converted, where in you example, its id from which it's being converted. 
With updated function signature, the implementation can mean both, the id of in_balance, or the id of out_balance.

Reply:
> With updated function signature, the implementation can mean both, the id of in_balance, or the id of out_balance.

IMO, that should be fine and expected of such a generic trait. Otherwise, its name should be more restricted (e.g. Option B) or it should provide a `from_asset_balance` (Option A).

Reply:
@tonyalaribe what do you think?

Reply:
We already have the `Convert` trait. I would also stick to the old name `to_asset_balance`. But yeah, maybe the trait should be renamed.

Reply:
So you are preferring Option B @bkchr? We need an opposite trait/function `from_asset_balance` for a #13608 which is a dependency of #13604

Reply:
Yeah Option B. Generally the name of the trait `BalanceConversion` and then appears `Asset` in the function names is weird, but naming is always hard and I don't have a better idea :P `from_asset_balance` also sounds reasonable.